component/src/main/java/org/wso2/carbon/event/input/adapter/mqtt/internal/util/MQTTAdapterListener.java

```diff
@@ -0,0 +1,234 @@
+        } catch (CertificateException e) {
+            throw new InputEventAdapterRuntimeException("TrustStore is not specified. So Security certificate" +
+                                                                " Exception happened.  ", e);
+        } catch (NoSuchAlgorithmException e) {
+            throw new InputEventAdapterRuntimeException("Algorithm is not available in KeyManagerFactory class.", e);
+        } catch (KeyStoreException e) {
+            throw new InputEventAdapterRuntimeException("Error in TrustStore Type", e);
+        } catch (KeyManagementException e) {
+            throw new InputEventAdapterRuntimeException("Error in Key Management", e);
+        } finally {
+            if (fileInputStream != null) {
+                fileInputStream.close();
+            }
+        }
+    }
+
+    public void startListener() throws MqttException {
+        // Connect to the MQTT server
+        mqttClient.connect(connectionOptions);
+
+        // Subscribe to the requested topic
+        // The QoS specified is the maximum level that messages will be sent to the client at.
+        // For instance if QoS 1 is specified, any messages originally published at QoS 2 will
+        // be downgraded to 1 when delivering to the client but messages published at 1 and 0
+        // will be received at the same level they were published at.
+        mqttClient.subscribe(topic);
+    }
+
+    public void stopListener(String adapterName) {
+        if (connectionSucceeded) {
+            try {
+                // Un-subscribe accordingly and disconnect from the MQTT server.
+                if (!ServerStatus.getCurrentStatus().equals(
+                        ServerStatus.STATUS_SHUTTING_DOWN) || cleanSession) {
+                    mqttClient.unsubscribe(topic);
+                }
+                mqttClient.disconnect(3000);
+            } catch (MqttException e) {
+                log.error("Can not unsubscribe from the destination " + topic
+                                  + " with the event adapter " + adapterName, e);
+            }
+        }
+        //This is to stop all running reconnection threads
+        connectionSucceeded = true;
+    }
+
+    @Override
+    public void connectionLost(Throwable throwable) {
+        log.warn("MQTT connection not reachable " + throwable);
+        connectionSucceeded = false;
+        new Thread(this).start();
+    }
+
+    @Override
+    public void messageArrived(String s, MqttMessage mqttMessage) throws Exception {
+        try {
+            String msgText = mqttMessage.toString();
+            if (log.isDebugEnabled()) {
+                log.debug(msgText);
+            }
+            PrivilegedCarbonContext.startTenantFlow();
+            PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantId);
+
+            if (log.isDebugEnabled()) {
+                log.debug("Event received in MQTT Event Adapter - " + msgText);
```

Comment:
We are logging the same msgText in the previous debug log. Why are we logging the same thing again without any modifications